pkg/helper/webconsole.go

```diff
@@ -0,0 +1,121 @@
+package helper
+
+import (
+	"context"
+	"fmt"
+	"github.com/RHsyseng/operator-utils/pkg/logs"
+	"github.com/RHsyseng/operator-utils/pkg/resource/read"
+	"github.com/RHsyseng/operator-utils/pkg/utils/kubernetes"
+	consolev1 "github.com/openshift/api/console/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+)
+
+var logger = logs.GetLogger("openshift-webconsole")
+
+// ConsoleLinkText is the text of the consoleLink shown on the webconsole
+const ConsoleLinkText = "APIManager - 3scale"
+
+// ConsoleLinkNamePrefix is the prefix applied to Console link for system master
+const ConsoleLinkNamePrefix = "system-master-link"
+
+// ConsoleLinkSupported checks if a ConsoleLink CRD exists
+func ConsoleLinkSupported() error {
+	gvk := schema.GroupVersionKind{Group: "console.openshift.io", Version: "v1", Kind: "ConsoleLink"}
+	return kubernetes.CustomResourceDefinitionExists(gvk)
+}
+
+// ReconcileConsoleLink creates a ConsoleLink object if it doesn't already exist
+func ReconcileConsoleLink(ctx context.Context, c client.Client, req *reconcile.Request) error {
+	route := getRoute(ctx, c, req)
+	if route == nil {
+		removeConsoleLink(ctx, c, req)
```

Comment:
Here if it fails to remove we should also requeue. There's an exception in this case: If it fails to remove because it does not exist anymore then we don't need to requeue

Reply:
added requeue